src/tests/test_transform.py

```diff
@@ -2,15 +2,115 @@
+nvdaVersion2021_1 = VersionCompatibility(V_2021_1, V_2021_1, V_2021_1)
+nvdaVersion2021_1_1 = VersionCompatibility(V_2021_1_1, V_2021_1, V_2021_1)
+nvdaVersion2022_1 = VersionCompatibility(V_2022_1, V_2022_1, V_2022_1)
 
-class Test_testExample(unittest.TestCase):
-	def test_example_true(self):
-		self.assertTrue(True)
 
-	def test_example_pass(self):
-		pass
+class Test_isAddonCompatible(unittest.TestCase):
+	def test_valid_with_api(self):
+		"""Confirm an addon is compatible with a fully tested API"""
+		addon = MockAddon()
+		addon.minNVDAVersion = V_2020_1
+		addon.lastTestedVersion = V_2020_2
+		self.assertTrue(_isAddonCompatible(addon, nvdaVersion2020_2))
 
-	# def test_example_fail(self):
-	# 	self.assertTrue(False, "Example test failure")
+	def test_valid_with_backwards_compatible_api(self):
+		"""Confirm an addon is compatible with a backwards compatible API"""
+		addon = MockAddon()
+		addon.minNVDAVersion = V_2020_1
+		addon.lastTestedVersion = V_2020_2
+		self.assertTrue(_isAddonCompatible(addon, nvdaVersion2020_3))
+
+	def test_not_valid_because_breaking_api(self):
+		"""Confirm an addon is not compatible with a breaking API"""
+		addon = MockAddon()
+		addon.minNVDAVersion = V_2020_1
+		addon.lastTestedVersion = V_2020_2
+		self.assertFalse(_isAddonCompatible(addon, nvdaVersion2021_1))
+
+	def test_not_valid_because_old_api(self):
+		"""Confirm an addon is not compatible with an old API"""
+		addon = MockAddon()
+		addon.minNVDAVersion = V_2021_1
+		addon.lastTestedVersion = V_2021_2
+		self.assertFalse(_isAddonCompatible(addon, nvdaVersion2020_3))
+
+
+class Test_getLatestAddons(unittest.TestCase):
+	def test_beta_stable(self):
+		"""Ensure addons are added to the correct channels"""
+		NVDAVersions = (nvdaVersion2020_2, nvdaVersion2020_3)
+		betaAddon = MockAddon()
+		betaAddon.minNVDAVersion = V_2020_1
+		betaAddon.lastTestedVersion = V_2020_3
+		betaAddon.channel = "beta"
+		betaAddon.addonVersionNumber = AddonVersion(0, 2, 1)
+		stableAddon = MockAddon()
+		stableAddon.minNVDAVersion = V_2020_1
+		stableAddon.lastTestedVersion = V_2020_2
+		stableAddon.channel = "stable"
+		stableAddon.addonVersionNumber = AddonVersion(0, 2)
+		self.assertDictEqual(getLatestAddons([betaAddon, stableAddon], NVDAVersions), {
+			V_2020_2: {"beta": {betaAddon.addonId: betaAddon}, "stable": {stableAddon.addonId: stableAddon}},
+			V_2020_3: {"beta": {betaAddon.addonId: betaAddon}, "stable": {stableAddon.addonId: stableAddon}},
+		})
+
+	def test_onlyNewerUsed(self):
+		"""Ensure only the newest addon is used for a version+channel"""
+		NVDAVersions = (nvdaVersion2020_2, nvdaVersion2020_3)
+		newAddon = MockAddon()
+		newAddon.addonId = "foo"
+		newAddon.minNVDAVersion = V_2020_1
+		newAddon.lastTestedVersion = V_2020_3
+		newAddon.channel = "beta"
+		newAddon.addonVersionNumber = AddonVersion(0, 2)
+		oldAddon = MockAddon()
+		oldAddon.addonId = "foo"
+		oldAddon.minNVDAVersion = V_2020_1
+		oldAddon.lastTestedVersion = V_2020_2
```

Comment:
If this is about testing version number comparisons, keep the other values constant. Ideally `newAddon` would be a copy of `oldAddon`, and just update the version number.

EG
```
		NVDAVersions = (nvdaVersion2020_3) # only one version necessary to demonstrate 
		oldAddon = MockAddon(
			channel = "beta"
			addonId = "foo",
			addonVersionNumber = AddonVersion(0, 1),
			minNVDAVersion = V_2020_1,
			lastTestedVersion = V_2020_3,
		)
		newAddon = copy(oldAddon)
		# This version should supercede 0.3 for nvdaVersion2020_3
		newAddon.addonVersionNumber = AddonVersion(0, 2)
```

Reply:
Good idea, I've improved this now.